src/eth/TransactionObject.js

```diff
@@ -121,7 +121,7 @@ export default class TransactionObject extends TransactionLifeCycle {
       if (receipt) return receipt;
 
       log(`Receipt is null. Retrying ${retries} more time(s)`);
-      return promiseWait((6 - retries) * 1500).then(() =>
+      return promiseWait((16 - retries) * 1500).then(() =>
```

Comment:
i shouldn't have hardcoded this originally. the right way to do this would be:
```js
_waitForReceipt(retries = 15, currentTry = 1) {
  // ...
  return promiseWait(currentTry * 1500).then(() =>
    this._waitForReceipt(retries - 1, currentTry + 1)
  );
  // ...